simplerunner.lua

```diff
@@ -315,20 +332,62 @@ function runner.run_benchmark_list(benchmarks, count, options)
 
     local package_path = package.path
     local package_cpath = package.cpath
-    
+    local results = {}
+
     for i, name in ipairs(benchmarks) do
-        local times = runner.runbench(name, count, options.scaling or scaling[name])
-        local stats = runner.calculate_stats(times)
-        print(string.format("  Mean: %f +/- %f, min %f, max %f", stats.mean, stats.cinterval, stats.min, stats.max))
-        if jitstats then
-            jitstats.print()
+        local scaling = options.scaling or scaling[name] or 1
+        local stats
+
+        if options.inprocess then
+            local times, jstats = runner.runbench(name, count, scaling)
+            stats = runner.calculate_stats(times)
+            stats.times = times
+            if jitstats then
+                stats.jitstats = jstats
+                jitstats.print()
+            end
+        else
+           runner.runbench_outprocess(name, count, scaling, options)
         end
-        -- Try to clean away the current benchmark and its data, so the behaviour of the GC is more predictable for the next benchmark.
-        collectgarbage("collect")
 
-        -- Restore the Lua module search path since we set it to the directory of the benchmark when we load it.
-        package.path = package_path
-        package.cpath = package_cpath
+        if options.inprocess then
+            -- Try to clean away the current benchmark and its data, so the behaviour of the GC is more predictable for the next benchmark.
+            collectgarbage("collect")
+
+            -- Restore the Lua module search path since we set it to the directory of the benchmark when we load it.
+            package.path = package_path
+            package.cpath = package_cpath
+        end
+    end
+end
+
+function runner.runbench_outprocess(name, count, scaling, options)
+    local optionsfile = "lastrun.json"
+    local cmdret = os.execute(string.format('%s %s @RUNBENCH@ "%s" %d "%s" "%s"', arg[-1], arg[0], name, count, scaling, optionsfile))
```

Comment:
This isn't a safe way of calling a child process. We need to use `execve` or equivalent.